core/src/com/fallenflame/game/LightInputProcessor.java

```diff
@@ -0,0 +1,90 @@
+package com.fallenflame.game;
+
+import com.badlogic.gdx.InputProcessor;
+import com.fallenflame.game.GameEngine;
+
+import static com.badlogic.gdx.scenes.scene2d.actions.Actions.delay;
+
+/** Controller to handle event-based inputs from the game. This is in a separate file from InputController, as
+ * InputController handles polling events (or events that occur on an update
+ * This is also separate from the LoadingMode InputProcessor to prevent LoadingMode from needing GameEngine or having to
+ * call a function from GDXRoot (therefore causing a cycle)
+ * Additionally, it splits the features well, and makes it easier to make a distinction between event-driven inputs
+ * that LoadingMode needs versus event-driven events that the main game needs*/
+
+public class LightInputProcessor implements InputProcessor {
+   private GameEngine gameEngine;
+
+   public LightInputProcessor(GameEngine g){
+       gameEngine = g;
+   }
+   /**Below are functions that are required to be implemented by InputProcessor. All return false if unused to indicate
+    * that the event was not handled by this inputProcessor. All return that return false here
```

Comment:
"All that return false here" very minor type

Reply:
lmao oops